### PR TITLE
Implement Pester tests for Android SDKs

### PR DESCRIPTION
**Pull request by vsafonkin**
# Description
- Moved list of Android software to toolset json file
- Added Pester tests for Android software

#### Related issue:
[#805](https://github.com/actions/virtual-environments-internal/issues/805)

## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/win/scripts/Tests/Android.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    $addonsTestCases = @()
+    $addonsList = $android.addon_list
+    $addonsList | ForEach-Object {
+        $addonsTestCases += @{ addonPackage = $_; installedPackages = $androidInstalledPackages }
+    }
+
+    It "Platform version <platformVersion> is installed" -TestCases $platformTestCases {
+        $regex = "platforms;$platformVersion"
+        [boolean]($installedPackages | Where-Object { $_ -match $regex }) | Should -Be $true
```
Comment (al-cheb) [inline images/win/scripts/Tests/Android.Tests.ps1:33]: Use built-in parameter: `Should -BeTrue`

File: images/win/scripts/Installers/Update-AndroidSDK.ps1 [hunk 2]
```diff
@@ -34,82 +34,45 @@ Expand-Archive -Path .\android-sdk-licenses.zip -DestinationPath 'C:\Program Fil
...
-Pop-Location
+# Get android content from toolset
+$android = (Get-ToolsetContent).android
+
+$sdkRoot = "C:\Program Files (x86)\Android\android-sdk"
+$sdkManager = "$sdkRoot\tools\bin\sdkmanager.bat"
+
+& $sdkManager --sdk_root=$sdkRoot "platform-tools"
+
+foreach ($platform in $android.platform_list) {
```
Comment (al-cheb) [inline images/win/scripts/Installers/Update-AndroidSDK.ps1:45]: Can we wrap in a simple function foreach block?

File: images/win/scripts/Tests/Android.Tests.ps1 (hunk 1 above)
Comment (vsafonkin) [reply images/win/scripts/Tests/Android.Tests.ps1]: Done

File: images/win/scripts/Installers/Update-AndroidSDK.ps1 [hunk 3]
```diff
@@ -34,82 +34,45 @@ Expand-Archive -Path .\android-sdk-licenses.zip -DestinationPath 'C:\Program Fil
...
-    }
-
-Pop-Location
+# Get android content from toolset
+$android = (Get-ToolsetContent).android
+
+$sdkRoot = "C:\Program Files (x86)\Android\android-sdk"
+$sdkManager = "$sdkRoot\tools\bin\sdkmanager.bat"
+
+& $sdkManager --sdk_root=$sdkRoot "platform-tools"
```
Comment (vmapetr) [inline images/win/scripts/Installers/Update-AndroidSDK.ps1:43]: `& $sdkManager --sdk_root=$sdkRoot` reused multiple times and can be moved to separate function.

File: images/win/toolsets/toolset-2019.json [hunk 4]
```diff
@@ -121,5 +121,75 @@
                 "4.3.0"
             ]
         }
-    ]
+    ],
+    "android": {
```
Comment (maxim-lobanov) [inline images/win/toolsets/toolset-2019.json:125]: Let's put all these arrays in line
```
"platform_list": [
            "android-30", "android-29", "android-28", "android-27", "android-25", "android-24", "android-23", "android-26", "android-22", "android-21", "android-19"
],
```
Current approach takes too much lines and decrease readability

File: images/win/scripts/Installers/Update-AndroidSDK.ps1 (hunk 2 above)
Comment (vsafonkin) [reply images/win/scripts/Installers/Update-AndroidSDK.ps1]: Done

File: images/win/scripts/Installers/Update-AndroidSDK.ps1 (hunk 3 above)
Comment (vsafonkin) [reply images/win/scripts/Installers/Update-AndroidSDK.ps1:43]: Done

File: images/win/toolsets/toolset-2019.json (hunk 4 above)
Comment (vsafonkin) [reply images/win/toolsets/toolset-2019.json:125]: Done

File: images/win/scripts/Installers/Update-AndroidSDK.ps1 [hunk 5]
```diff
@@ -34,82 +34,49 @@ Expand-Archive -Path .\android-sdk-licenses.zip -DestinationPath 'C:\Program Fil
...
-        setx ANDROID_NDK_HOME $ndk_root /M
-        setx ANDROID_NDK_PATH $ndk_root /M
-    } else {
-        Write-Host "NDK is not installed at path $ndk_root"
-        exit 1
-    }
-
-Pop-Location
+# Get android content from toolset
+$android = (Get-ToolsetContent).android
```
Comment (AlenaSviridenko) [inline images/win/scripts/Installers/Update-AndroidSDK.ps1:38]: may be it it better to rename it to $toolset, for exmple, so later we could you it like `$toolset.platform_list`, `$toolset.build_tools`? We are in the Android file already, and name `$android` seems very generic

Comment (vsafonkin) [reply images/win/scripts/Installers/Update-AndroidSDK.ps1]: Renamed to `$androidToolset` :)

File: images/win/scripts/Tests/Android.Tests.ps1 [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    $addonsTestCases = @()
+    $addonsList = $androidToolset.addon_list
+    $addonsList | ForEach-Object {
+        $addonsTestCases += @{ addonPackage = $_; installedPackages = $androidInstalledPackages }
+    }
+
+    It "Platform version <platformVersion> is installed" -TestCases $platformTestCases {
+        $regex = "platforms;$platformVersion"
+        [boolean]($installedPackages | Where-Object { $_ -match $regex }) | Should -BeTrue
```
Comment (al-cheb) [inline images/win/scripts/Tests/Android.Tests.ps1:33]: Does it make sense to use -Match parameter?

`$installedPackages | Should -Match "platforms;$platformVersion"`

File: images/win/toolsets/toolset-2016.json [hunk 7]
```diff
@@ -112,5 +112,34 @@
...
+            "extras;android;m2repository",
+            "extras;google;m2repository",
+            "extras;google;google_play_services",
+            "extras;m2repository;com;android;support;constraint;constraint-layout-solver;1.0.2",
+            "extras;m2repository;com;android;support;constraint;constraint-layout-solver;1.0.1",
+            "extras;m2repository;com;android;support;constraint;constraint-layout;1.0.2",
+            "extras;m2repository;com;android;support;constraint;constraint-layout;1.0.1"
+        ],
+        "addon_list": [
+            "add-ons;addon-google_apis-google-24",
```
Comment (MaksimZhukov) [inline images/win/toolsets/toolset-2016.json:133]: Why did we add the `add-ons` prefix here and skip it in the `toolset-2019.json` file?

File: images/win/toolsets/toolset-2016.json [hunk 8]
```diff
@@ -112,5 +112,34 @@
                 "4.3.0"
             ]
         }
-    ]
+    ],
+    "android": {
+        "platform_list": [
+            "android-30", "android-29", "android-28", "android-27", "android-25", "android-24", "android-23", "android-26", "android-22", "android-21", "android-19"
```
Comment (MaksimZhukov) [inline images/win/toolsets/toolset-2016.json:118]: incorrect order: `... "android-23", "android-26", "android-22", ...`

File: images/win/toolsets/toolset-2016.json (hunk 7 above)
Comment (MaksimZhukov) [reply images/win/toolsets/toolset-2016.json:133]: The same for the `extras` prefix

File: images/win/scripts/Tests/Android.Tests.ps1 (hunk 6 above)
Comment (vsafonkin) [reply images/win/scripts/Tests/Android.Tests.ps1]: Done, thank you!

File: images/win/toolsets/toolset-2016.json (hunk 8 above)
Comment (vsafonkin) [reply images/win/toolsets/toolset-2016.json]: Fixes

File: images/win/toolsets/toolset-2016.json (hunk 7 above)
Comment (vsafonkin) [reply images/win/toolsets/toolset-2016.json]: Fixed, thank you!

File: images/win/scripts/Tests/Android.Tests.ps1 [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+        $platformTestCases += @{ platformVersion = $_; installedPackages = $androidInstalledPackages }
+    }
+
+    $buildToolsTestCases = @()
+    $buildToolsList = $androidToolset.build_tools
+    $buildToolsList | ForEach-Object {
+        $buildToolsTestCases += @{ buildToolsVersion = $_; installedPackages = $androidInstalledPackages }
+    }
+
+    It "Platform version <platformVersion> is installed" -TestCases $platformTestCases {
```
Comment (maxim-lobanov) [inline images/win/scripts/Tests/Android.Tests.ps1:19]: I think we also need to validate `extra_list`, `addon_list`, `additional_tools`

Comment (vsafonkin) [reply images/win/scripts/Tests/Android.Tests.ps1:25]: Fixed

**Pull request merged**